spec/unit/transaction/report_spec.rb

```diff
@@ -136,6 +136,36 @@
     expect(report.metrics['time'].values.any? {|metric| metric.first =~ /whit/i}).to be_falsey
   end
 
+  describe "when exclude_unchanged_resources is true" do
+    before do
+      @test_dir = Dir.mktmpdir
+    end
+
+    def generate_report_and_get_resource_statuses(test_dir)
+      Puppet[:exclude_unchanged_resources] = true
+      file = Puppet::Type.type(:file).new(:path => test_dir, :ensure => 'directory')
+      catalog = Puppet::Resource::Catalog.new
+      catalog.add_resource(file)
+      report = Puppet::Transaction::Report.new
+      catalog.apply(:report => report)
+      report.finalize_report
```

Comment:
You may be able to replace the catalog construction and application with https://github.com/puppetlabs/puppet/blob/e82968bf46377f9f604d745026270ea5d709fb65/spec/lib/puppet_spec/compiler.rb#L31

Something like:

```ruby
transaction = apply_compiled_manifest(<<END)
file { '#{test_dir}':
  ensure => directory
}
END
report = transaction.report
...
```
You might need to `include PuppetSpec::Compiler` at the top of the describe like we do for `PuppetSpec::Files`